src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedPrivateImplementationDetailsType.cs

```diff
@@ -0,0 +1,189 @@
+
+        internal override bool IsInterpolatedStringHandlerType => false;
+
+        internal override bool HasSpecialName => _privateImplementationDetails.IsSpecialName;
+
+        internal override bool IsComImport => false;
+
+        internal override bool IsWindowsRuntimeImport => false;
+
+        internal override bool ShouldAddWinRTMembers => false;
+
+        internal override TypeLayout Layout => new TypeLayout(_privateImplementationDetails.Layout, (int)_privateImplementationDetails.SizeOf, (byte)_privateImplementationDetails.Alignment);
+
+        internal override CharSet MarshallingCharSet => _privateImplementationDetails.StringFormat;
+
+        internal override bool HasDeclarativeSecurity => false;
+
+        internal override bool IsInterface => false;
+
+        internal override NamedTypeSymbol? NativeIntegerUnderlyingType => null;
+
+        internal override NamedTypeSymbol BaseTypeNoUseSiteDiagnostics => _objectType;
+
+        internal override bool IsRecord => false;
+
+        internal override bool IsRecordStruct => false;
+
+        internal override ObsoleteAttributeData? ObsoleteAttributeData => null;
+
+        public override ImmutableArray<Symbol> GetMembers() => ImmutableArray<Symbol>.Empty;
+
+        public override ImmutableArray<Symbol> GetMembers(string name) => ImmutableArray<Symbol>.Empty;
+
+        public override ImmutableArray<NamedTypeSymbol> GetTypeMembers() => ImmutableArray<NamedTypeSymbol>.Empty;
+
+        public override ImmutableArray<NamedTypeSymbol> GetTypeMembers(ReadOnlyMemory<char> name, int arity) => ImmutableArray<NamedTypeSymbol>.Empty;
+
+        public override ImmutableArray<NamedTypeSymbol> GetTypeMembers(ReadOnlyMemory<char> name) => ImmutableArray<NamedTypeSymbol>.Empty;
+
+        protected override NamedTypeSymbol WithTupleDataCore(TupleExtraData newData)
+        {
+            throw ExceptionUtilities.Unreachable();
+        }
+
+        internal override NamedTypeSymbol AsNativeInteger()
+        {
+            throw ExceptionUtilities.Unreachable();
+        }
+
+        internal override ImmutableArray<string> GetAppliedConditionalSymbols() => ImmutableArray<string>.Empty;
+
+        internal override AttributeUsageInfo GetAttributeUsageInfo()
+        {
+            throw ExceptionUtilities.Unreachable();
+        }
+
+        internal override NamedTypeSymbol GetDeclaredBaseType(ConsList<TypeSymbol> basesBeingResolved) => _objectType;
+
+        internal override ImmutableArray<NamedTypeSymbol> GetDeclaredInterfaces(ConsList<TypeSymbol> basesBeingResolved) => ImmutableArray<NamedTypeSymbol>.Empty;
+
+        internal override ImmutableArray<Symbol> GetEarlyAttributeDecodingMembers() => ImmutableArray<Symbol>.Empty;
+
+        internal override ImmutableArray<Symbol> GetEarlyAttributeDecodingMembers(string name) => ImmutableArray<Symbol>.Empty;
+
+        internal override IEnumerable<FieldSymbol> GetFieldsToEmit() => SpecializedCollections.EmptyEnumerable<FieldSymbol>();
```

Comment:
>SpecializedCollections.EmptyEnumerable<FieldSymbol>() [](http://example.com/codeflow?start=72&length=53)

Aren't there fields in PrivateImplementationDetails? If there are fields, but this method should not be used, could we `throw Unreachable()`?